pkg/utils/kubeclient/pod.go

```diff
@@ -53,6 +57,42 @@ func IsFailedPod(pod *corev1.Pod) bool {
 	return pod != nil && pod.Status.Phase == corev1.PodFailed
 }
 
+// TailPodLogs tail pod's log.
+// Given name and namespace of pod, lines eq: "tail -n <lines>".
+func TailPodLogs(name, namespace string, lines int64) (logstr string, err error) {
+	err = initClient()
+	if err != nil {
+		return "", err
+	}
+	req := clientset.CoreV1().Pods(namespace).GetLogs(name, &corev1.PodLogOptions{TailLines: &lines})
+	podLogs, err := req.Stream(context.TODO())
+	if err != nil {
+		return "", err
+	}
+	defer podLogs.Close()
+
+	buf := new(bytes.Buffer)
+	_, err = io.Copy(buf, podLogs)
+	if err != nil {
+		return "", err
+	}
+	logstr = buf.String()
+	return
+}
+
+// ListPodsByLabel gets podList with given namespace and labels.
+func ListPodsByLabel(ctx context.Context, namespace string, label map[string]string) (podList *corev1.PodList, err error) {
```

Comment:
I think it's better to refactor the function with `client.List(&corev1.podList, metav1.ListOptions(...))`

Reply:
> I think it's better to refactor the function with `client.List(&corev1.podList, metav1.ListOptions(...))`

Got it!